src/UseCase/BannerSelection/CampaignConfigurationLoader.php

```diff
@@ -0,0 +1,117 @@
+<?php
+
+declare( strict_types = 1 );
+
+namespace WMDE\BannerServer\UseCase\BannerSelection;
+
+use Psr\Log\LoggerInterface;
+use Symfony\Component\Yaml\Exception\ParseException;
+use Symfony\Component\Yaml\Yaml;
+use WMDE\BannerServer\Entity\BannerSelection\Banner;
+use WMDE\BannerServer\Entity\BannerSelection\Bucket;
+use WMDE\BannerServer\Entity\BannerSelection\Campaign;
+use WMDE\BannerServer\Entity\BannerSelection\CampaignCollection;
+use WMDE\BannerServer\Utils\SystemRandomIntegerGenerator;
+
+/**
+ * @license GNU GPL v2+
+ */
+class CampaignConfigurationLoader {
+
+	private $configFile;
+	private $logger;
+
+	public function __construct( LoggerInterface $logger, string $configFile ) {
+		$this->configFile = $configFile;
+		$this->logger = $logger;
+	}
+
+	public function getCampaignCollection(): CampaignCollection {
+		$campaigns = [];
+		foreach ( $this->parseConfiguration() as $campaignName => $campaignData ) {
+			try {
+				$campaign = $this->buildCampaignFromData( $campaignName, $campaignData );
+				$campaigns[] = $campaign;
+			}
+			catch ( \Exception $e ) {
```

Comment:
To guard against other exceptions that might happen (although very unlikely in this case), you should create your own exception class. 

Since we're using this internally, we can get away with the general exception. Though using a `DomainException` would make it a bit more specific.